contracts/src/Teleporter/upgrades/TeleporterRegistry.sol

```diff
@@ -0,0 +1,69 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import "../../WarpProtocolRegistry.sol";
+import "../ITeleporterMessenger.sol";
+
+/**
+ * @dev TeleporterRegistry contract is a {WarpProtocolRegistry} and provides an upgrade
+ * mechanism for teleporter messenger contracts.
+ */
+contract TeleporterRegistry is WarpProtocolRegistry {
+    mapping(address => uint256) private _addressToVersion;
+
+    constructor(
+        uint256[] memory initialVersions,
+        address[] memory initialProtocolAddresses
+    ) WarpProtocolRegistry(initialVersions, initialProtocolAddresses) {}
+
+    /**
+     * @dev Gets the {ITeleporterMessenger} contract of the given `version`.
+     */
+    function getTeleporterFromVersion(
+        uint256 version
+    ) external view returns (ITeleporterMessenger) {
+        return ITeleporterMessenger(_getAddressFromVersion(version));
+    }
+
+    /**
+     * @dev Gets the latest {ITeleporterMessenger} contract.
+     */
+    function getLatestTeleporter()
+        external
+        view
+        returns (ITeleporterMessenger)
+    {
+        return ITeleporterMessenger(_getAddressFromVersion(_latestVersion));
+    }
+
+    /**
+     * @dev Gets the version of the given `protocolAddress`.
+     * If `protocolAddress` is not a valid protocol address, returns 0.
```

Comment:
Is 0 a valid version? Do we need to differentiate between "not valid protocol address" and version 0 of a valid address?

Reply:
0 is not a valid version as the `_latestVersion` is iniitialized as 0, and we can only add protocol with version `_latestVersion+1`. Added some more comments to help note this, do you think a constant `INVALID_VERSION = 0` would help make more obvious?

Reply:
Extra comments should be enough. It is clear from the rest of the code that 0 is invalid, but it is not clear when just looking at this function.